src/odemis/dataio/csv.py

```diff
@@ -45,159 +45,260 @@ def export(filename, data):
+    elif dims == "TC" and data.ndim == 2:
         logging.debug("Exporting temporal spectrum data to CSV")
-        if acq_type != model.MD_AT_TEMPSPECTRUM:
-            logging.warning("Data seems to be temporal spectrum, but acq_type is %s", acq_type)
-
-        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
-        if unit_c == "m":
-            spectrum_range = [s * 1e9 for s in spectrum_range]
-            unit_c = "nm"
-        time_range, unit_t = spectrum.get_time_range(data)
-        if unit_t == "s":
-            time_range = [t * 1e12 for t in time_range]
-            unit_t = "ps"
-
-        headers = ["time(" + unit_t + ")\\wavelength(" + unit_c + ")"] + spectrum_range
-        rows = [(t,) + tuple(d) for t, d in zip(time_range, data)]
-
-        with open(filename, 'w', newline='') as fd:
-            csv_writer = csv.writer(fd)
-            csv_writer.writerow(headers)
-            csv_writer.writerows(rows)
-
-    elif data.ndim == 2 and dims == "AC":
-        logging.debug("Exporting angular spectrum data to CSV")
-        if acq_type != model.MD_AT_EK:
-            logging.warning("Data seems to be EK, but acq_type is %s", acq_type)
+        _export_temporal_spectrum_data(acq_type, data, filename)
 
-        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
-        if unit_c == "m":
-            spectrum_range = [s * 1e9 for s in spectrum_range]
-            unit_c = "nm"
+    # Export Angular Spectrum Data
+    elif dims == "AC" and data.ndim == 2:
+        logging.debug("Exporting angular spectrum data to CSV")
+        _export_angular_spectrum_data(acq_type, data, filename)
 
-        angle_range, unit_a = spectrum.get_angle_range(data)
-        if unit_a == "rad":
-            unit_a = "°"
-            angle_range = [math.degrees(theta) for theta in angle_range]  # Convert radians to degrees
-            assert len(angle_range) == len(data)
+    # Export AR Data
+    elif acq_type == model.MD_AT_AR:
+        logging.debug("Exporting AR data to CSV")
+        _export_ar_data(data, filename)
 
-        headers = ["angle(" + unit_a + ")\\wavelength(" + unit_c + ")"] + spectrum_range
-        rows = [(t,) + tuple(d) for t, d in zip(angle_range, data)]
+    else:
+        raise ValueError(f"Unknown acquisition type {acq_type} of data (dims = {dims}) to be exported as CSV")
 
-        with open(filename, 'w') as fd:
-            csv_writer = csv.writer(fd)
-            csv_writer.writerow(headers)
-            csv_writer.writerows(rows)
 
+def _export_spectrum_data(acq_type, data, filename):
+    """
+    This will export the spectrum data to a CSV file using the type of microscope,
+    the acquired data and filename title in its creation.
+    :param acq_type: data.metadata.get(model.MD_ACQ_TYPE, None)
+    :param data: (model.DataArray)
+    :param filename: (unicode)
+    """
+
+    if acq_type != model.MD_AT_SPECTRUM:
+        logging.warning("Data seems to be spectrum, but acq_type is %s", acq_type)
```

Comment:
I think it's strange that not all functions have the same signature. Let's move this check back into the `if/elif` block, and drop the `acq_type` argument`. This way all the functions would take `(data, filename)` as arguments, only.

Reply:
I totally agree with that, could you point out to me for which acquisition type I have to do a check regarding to data type?

Reply:
I think that all the acquisition types possible to check are already in the code. The other type of export (eg, angle data only) doesn't match an existing acquisition type (they are obtained by processing data from an acquisition, not a direct acquisition).